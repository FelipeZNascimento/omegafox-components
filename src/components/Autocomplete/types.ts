export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
