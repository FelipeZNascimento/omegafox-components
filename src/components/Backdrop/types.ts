export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
