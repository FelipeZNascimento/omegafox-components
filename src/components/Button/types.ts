export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
