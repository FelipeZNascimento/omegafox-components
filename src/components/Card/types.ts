export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
