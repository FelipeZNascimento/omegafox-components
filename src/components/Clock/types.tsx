import {
  FOOTBALL_MATCH_STATUS,
  AMERICAN_FOOTBALL_MATCH_STATUS
} from './constants';
import { TBetValues } from '../types';

type TClock = {
  time: number;
};

type FootballKeys = keyof typeof FOOTBALL_MATCH_STATUS;
export type TClockFootball = TClock & {
  status: typeof FOOTBALL_MATCH_STATUS[FootballKeys];
};

type AmericanFootballKeys = keyof typeof AMERICAN_FOOTBALL_MATCH_STATUS;
export type TClockAmericanFootball = TClock & {
  status: typeof AMERICAN_FOOTBALL_MATCH_STATUS[AmericanFootballKeys];
};

export interface IClockProps {
  betValue: TBetValues | null;
  clock: TClockAmericanFootball | TClockFootball;
  isMatchStarted?: boolean;
  isExpandable: boolean;
  isExpanded: boolean;
  isMatchEditable: boolean;
  timestamp: number;
}

export interface IClockMobileProps {
  clock: TClockAmericanFootball | TClockFootball;
  timestamp: number;
}
