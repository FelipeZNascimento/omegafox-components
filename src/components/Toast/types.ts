export interface IToastProps {
  text: string;
  variant?: 'error' | 'success' | 'neutral';
  onClose: () => void;
}
