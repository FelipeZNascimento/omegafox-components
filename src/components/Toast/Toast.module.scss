@import "../../styling/alignments.scss";
@import "../../styling/colors.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.container {
  @include padded(all, m);

  bottom: 4px;
  border-radius: $border-radius;
  left: 50%;
  -webkit-transform: translateX(-50%);
  transform: translateX(-50%);

  position: fixed;
  width: 90vw;
  z-index: 99999;

  .xButton {
    @include font-colored($color-grey2);
    @include font-sized(l2);
    @include padded(horizontal, s);

    top: 50%;
    transform: translateY(-50%);

    cursor: pointer;
    font-weight: bold;
    position: absolute;
    right: 4px;

    &:hover {
      @include font-colored($color-white);
    }
  }
}

.red {
  @include bg-rgba-colored($color-red, 0.9);
  @include font-colored($color-white);
}

.green {
  @include bg-rgba-colored($color-green, 0.9);
  @include font-colored($color-white);
}

.grey {
  @include bg-rgba-colored($color-grey4, 0.9);
  @include font-colored($color-white);
}
