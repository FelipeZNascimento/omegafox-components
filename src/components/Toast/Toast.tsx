import React from 'react';

import classNames from 'classnames';
import { IToastProps } from './types';
import styles from './Toast.module.scss';

export const Toast = ({ text, variant = 'neutral', onClose }: IToastProps) => {
  const containerClass = classNames({
    [styles.container]: true,
    [styles.grey]: variant === 'neutral',
    [styles.green]: variant === 'success',
    [styles.red]: variant === 'error'
  });

  return (
    <div className={containerClass}>
      {text}
      <div className={styles.xButton} onClick={onClose}>
        x
      </div>
    </div>
  );
};
