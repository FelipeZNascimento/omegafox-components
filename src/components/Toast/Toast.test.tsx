import React from 'react';
import { render } from '@testing-library/react';
import { Toast } from './Toast';
import { IToastProps } from './types';

describe('<Toast />', () => {
  const renderComponent = ({ text, variant }: IToastProps) =>
    render(<Toast text={text} variant={variant} onClose={jest.fn} />);

  it('should render', () => {
    const text = 'Click';
    const { container } = renderComponent({
      text: text,
      variant: 'error',
      onClose: jest.fn
    });

    expect(container).toMatchSnapshot();
  });
});
