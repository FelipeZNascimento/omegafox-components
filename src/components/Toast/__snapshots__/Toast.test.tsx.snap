// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> should render 1`] = `
<div>
  <div
    class="container red"
  >
    Click
    <div
      class="xButton"
    >
      x
    </div>
  </div>
</div>
`;
